CiscoESA/cisco-esa/ingest/parser.yml

```diff
@@ -0,0 +1,312 @@
+          parsed_event.message.deviceDirection: network.direction
+
+      - set:
+          host.name: "{{parsed_event.message.dvchost or {parsed_event.message.get('deviceDns Domain')}}"
+          host.id: "{{parsed_event.message.get('device ExternalId')}}"
+          log.syslog.facility.name: "{{parsed_event.message.deviceFacility}}"
+          observer.ingress.interface.name: "{{parsed_event.message.get('deviceInbound Interface')}}"
+          host.domain: "{{parsed_event.message.get('deviceNt Domain')}}"
+          observer.egress.interface.name: "{{parsed_event.message.get('Device Outbound Interface')}}"
+          process.name: "{{parsed_event.message.get('deviceProcess Name')}}"
+          server.nat.ip: "{{parsed_event.message.get('device Translated Address')}}"
+          destination.domain: "{{parsed_event.message.dhost or parsed_event.message.dntdom}}"
+          destination.port: "{{parsed_event.message.dpt}}"
+          destination.ip: "{{parsed_event.message.dst}}"
+          event.timezone: "{{parsed_event.message.dtz}}"
+          destination.user.id: "{{parsed_event.message.duid}}"
+          destination.user.name: "{{parsed_event.message.duser}}"
+          server.ip: "{{parsed_event.message.dvc}}"
+          destination.mac: "{{parsed_event.message.dmac}}"
+          event.end: "{{parse_end_date.date}}"
+          file.inode: "{{parsed_event.message.fileId}}"
+          file.mtime: "{{parsed_event.message.get('fileModification Time')}}"
+          file.path:  "{{parsed_event.message.filePath}}"
+          file.type:  "{{parsed_event.message.fileType}}"
+          host.network.ingress.bytes: "{{parsed_event.message.in}}"
+          host.network.egress.bytes: "{{parsed_event.message.out}}"
+          event.outcome: "{{parsed_event.message.outcome}}"
+          event.action: "{{parsed_event.message.reason}}"
+          url.original: "{{parsed_event.message.request}}"
+          user_agent.original: "{{parsed_event.message.get('requestClient Application')}}"
+          http.request.referrer: "{{parsed_event.message.requestContext}}"
+          http.request.method: "{{parsed_event.message.requestMethod}}"
+          host.hostname: "{{parsed_event.message.shost or parsed_event.message.sourceHostName}}"
+          source.domain: "{{parsed_event.message.get('sourceDns Domain')}}"
+          source.mac: "{{parsed_event.message.smac}}"
+          source.port: "{{parsed_event.message.spt}}"
+          source.nat.ip: "{{parsed_event.message.get('source Translated Address')}}"
+          source.nat.port: "{{parsed_event.message.get('source TranslatedPort')}}"
+          event.start:  "{{parse_start_date.date}}"
+          email.message_id: "{{parsed_event.message.ExternalMsgID}}"
+          email.local_id: "{{parsed_event.message.ESAMID}}"
+          rule.name: "{{parsed_event.message.MailPolicy}}"
+          email.attachments: "{{parsed_event.message.ESAAttachmentDetails}}"
+          url.full: "{{parsed_event.message.ESAURLDetails}}"
```

Comment:
We have to extract information from these fields.
However, I wonder if, currently, we are able to load their content !?